PowerPointLabs/PowerPointLabs/PositionsLab/PositionsPaneWPF.xaml.cs

```diff
@@ -1353,114 +1383,124 @@ public void ExecutePositionsAction(Action<List<PPShape>, bool> positionsAction,
 
         public void ExecutePositionsAction(Action<List<PPShape>, float> positionsAction, float dimension, bool isPreview)
         {
-            Selection selection = this.GetCurrentSelection();
-            if (!HandleInvalidSelection(isPreview, selection))
-            {
-                // invalid selection!
-                return;
-            }
-
-            ShapeRange simulatedShapes = null;
-
-            try
+            // Run twice to get the right operation
+            for (int noOfRuns = 0; noOfRuns < 2; noOfRuns++)
```

Comment:
I don't really understand why running twice will get the right operation. Could you explain it to me?

Reply:
@IanTeo its a PowerPoint thing. I determined it via manual testing and made sure it worked for different shapes/pictures. But ultimately I do not know why exactly running twice is needed.